public/ngapp/register/controllers.js

```diff
@@ -530,20 +533,65 @@ controllers.controller('AppsCtrl', ['$scope', '$rootScope', '$state', 'appsInfoF
 
       modalInstance = deleteModal.modalInstance(templateUrl, name, type, itemType);
       modalInstance.result.then(function (name) {
-        $scope.apps = _.filter($scope.apps, function (app) {
-          return app.App.Name != Name;
-        });
-        $scope.addAlert({
-            type: 'success',
-            message: "App '" + name + "' deleted successfully.",
-            icon: 'glyphicon glyphicon-ok'
+        appsInfoFactory.deleteApp(name, function(data){
+          $scope.apps = _.filter($scope.apps, function (app) {
+            return app != Name;
+          });
+          $scope.addAlert({
+              type: 'success',
+              message: "App '" + name + "' deleted successfully.",
+              icon: 'glyphicon glyphicon-ok'
+          });
+          $scope.name = "";
```

Comment:
Write a function or method to initialize these values:

```
         $scope.name = "";
         $scope.root = "";
         $scope.repo = "";
         $scope.email = "";
         $scope.internal = false;
         $scope.non_atlantis = false
```
Also, do the necessary changes in other part of code like `line 553 to 558` and few others.